src-java/floodlight-service/floodlight-api/src/main/java/org/openkilda/messaging/info/group/GroupDumpResponse.java

```diff
@@ -26,21 +29,38 @@
 import lombok.Value;
 
 import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
 
 @Value
 @EqualsAndHashCode(callSuper = false)
-public class GroupDumpResponse extends InfoData {
-    @JsonProperty("switch_id")
-    SwitchId switchId;
+public class GroupDumpResponse extends InfoData implements Chunkable<GroupDumpResponse> {
 
     @JsonProperty("group_speaker_data")
     List<GroupSpeakerData> groupSpeakerData;
 
     @JsonCreator
     @Builder
-    public GroupDumpResponse(@JsonProperty("switch_id") SwitchId switchId,
-                             @JsonProperty("group_speaker_data") List<GroupSpeakerData> groupSpeakerData) {
-        this.switchId = switchId;
+    public GroupDumpResponse(@JsonProperty("group_speaker_data") List<GroupSpeakerData> groupSpeakerData) {
         this.groupSpeakerData = groupSpeakerData;
     }
+
+    @Override
+    public List<GroupDumpResponse> split(int chunkSize) {
+        return Utils.split(groupSpeakerData, chunkSize).stream()
+                .map(GroupDumpResponse::new)
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Unites several responses into one.
+     */
+    public static GroupDumpResponse unite(List<GroupDumpResponse> dataList) {
```

Comment:
Can this be moved to Chunkable as default method?

Reply:
I wish I can, but after trying for several days to do it I failed. Base interface 
1. can't create new instance of child class
2. do not know how to each type of child. 

I tried to add non static unite method to override it in each class, but it bring more problems. 
If you have any idea - I'll be happy to hear it, because I'm run out of ideas :(

Reply:
Fair enough.